### DISCUSSION TITLE
What's here? (offline local search)

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#644] [Q&A] [answered]
"**What's here?**" is a new **offline** search function coming in next Cruiser:

- Click anywhere on the map and select "**What's here?**"

- **Offline** search checks for POIs in the selected area of the map

- One or more POIs will appear on the map with their details

- They can be used for route planning or information display

- The search distance can be selected in "Settings | Search | Advanced"

Everything works **offline** and extremely fast

<img src="https://github.com/user-attachments/assets/79876cef-995a-4c5e-9c11-adb3e3aaadbc" width="50%"/><img src="https://github.com/user-attachments/assets/ea3216f6-d617-4735-99c8-c12e95bbe1ef" width="50%"/>

**Comment by bios4** [comment]
Seems to be a nice addition, even though I don't know how often I personally will use it in reality.
So I would like to suggest not putting the new function first on these tiles.

My specific suggestion would be to show the "Show whole route" tile at the very first position, then the new "What is here?" option and then everything else in the list starting from "Show address".

**Comment by Ciutnik** [comment]
This is useful when you roughly know the location of a tour's via point (or destination) on the map, but don't want to create your own Favorite. I hope that thanks to this, instead of "VIA point #" we will be able to have somehow better displayed points.

I know and use a similar function in my other navigation app.

**Reply by 0709wiwiwi** [reply]
_I know and use a similar function in my other navigation app._
And that app is ?  Show us .... 
Let me guess a via point with a custom icon ?

**Reply by devemux86** [reply]
Google Maps has a "What's here" function.

**Reply by Ciutnik** [reply]
@0709wiwiwi 
> And that app is ?

NaviExpert (not based on OSM data). 
Outside my country the app is of little use, but here I find it unmatched when it comes to avoiding traffic jams. I use it when I just want to get from point A to point B efficiently in a city full of roadworks or events that require street closures.

There is a "Find near cursor" option here, with the ability to add found items to the route.

<img src="https://github.com/user-attachments/assets/08569bd6-78cd-44c1-a9d7-a6211da8b959" width="49%"/> <img src="https://github.com/user-attachments/assets/99c16bd4-72c5-4814-8a40-537a3f23eafa" width="49%"/>

**Reply by 0709wiwiwi** [reply]
@Ciutnik.
Your example.
Are the poi than considered via points that participate in the calculations ? 
Poi snapped to the nearest road as Via Points with a custom icon as shown ?

**Reply by Ciutnik** [reply]
@0709wiwiwi 
> Are the poi than considered via points that participate in the calculations ?

If I understood the question correctly, yes, you can add the selected POI to the route as its integral part, just like the address or coordinates. The route will be recalculated immediately.

> Poi snapped to the nearest road as Via Points with a custom icon as shown ?

Icons are shown only in the search list. As via points in navigation and on the map they are marked with the letters A, B, C... (with name but without any icon) - to indicate their order.
The application does not support shaping points as well as GPX files. 

---

As for the functionality: apart from the traditional "Find" (by name, address, coordinates...), it is possible to indicate a place on the map and select the "Find nearby" function, and then: 
- the application will search for a dozen POIs (also coordinates and addresses) and display them on the map as pins (photo 1),
- you can point to any of the pins or display a list of them (photo 2), including the category icon, name, address, distance,
- then you have the "Add to route" function,
- if it is the only point, it is set as a destination, if another one, it is added as a via point (in optimum order),
- you start the navigation, et voila.

**Reply by devemux86** [reply]
>it is possible to indicate a place on the map and select the "Find nearby" function, and then:
>- the application will search for a dozen POIs (also coordinates and addresses) and display them on the map as pins (photo 1),
>- you can point to any of the pins or display a list of them (photo 2), including the category icon, name, address, distance,
>- then you have the "Add to route" function,

All these functions will be available in Cruiser too.

---

We can discuss the search distance?

- It can start from a min distance and search with increasing distance until it finds one or more POIs.

- Or we can have a fixed distance setting and immediately find all POIs within the selected radius.

**Reply by Ciutnik** [reply]
It seems to me that the search should cover a circle no larger than 100-200 meters, but limited to 10-20 results. After all, we are looking for "What is HERE" 🙂

**Reply by devemux86** [reply]
Limited how? The first 10-20 nearest to the center?
Then users will ask to adjust the nearest number and the distance.

Probably the most customized solution is an adjustable distance.
Then everyone can select if want a small or large radius of search.

**Reply by Ciutnik** [reply]
> Probably the most customized solution is an adjustable distance.

Maybe instead of adjustable distance, correlate it with the currently displayed map area? Some _scale → distance_ converter (not necessarily linear)? The higher the zoom value, the fewer results to show 🙂 This would allow to avoid additional settings.

**Reply by devemux86** [reply]
This is already available when we press the search button and select "Nearby points of interest",
it searches offline within the visible boundaries of the map.

"What's here?" is a localized search, without the need to fully zoom in to search for what a particular location is.

**Comment by devemux86** [comment]
We can inspect all POIs on the map that now only show icon and name.
Or simply inspect what exactly is in any selected area, building, location.

Remember that the full OpenStreetMap database is available **offline** in Cruiser.
So we can display any POI data. Like we display name, address, opening hours.

**Reply by DolomitesTom** [reply]
Works very well, at the moment only the name is displayed when tapping on the icon. Yes, why not, show more infos like opening hours.

**Reply by devemux86** [reply]
It displays name, address, opening hours, if they exist in OpenStreetMap database.
For example see the image in the [first post](https://github.com/devemux86/cruiser/discussions/644#discussion-7340901).

It is the same displayed data with the offline search in other areas of Cruiser,
like with the search button.

**Comment by devemux86** [comment, marked as answer]
- What's here? (offline local search) is available in https://github.com/devemux86/cruiser/discussions/684
